spring-integration-file/src/main/java/org/springframework/integration/file/filters/ExpressionFileListFilter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.filters;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.util.Assert;
+
+/**
+ * A SpEL expression based {@link AbstractFileListFilter} implementation.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.0
+ */
+public class ExpressionFileListFilter<F> extends AbstractFileListFilter<F>
```

Comment:
Given that SpEL is not type safe, and we don't need subclasses, should this be generic?

    public class ExpressionFileListFilter extends AbstractFileListFilter<Object>

?